packages/aws-cdk-lib/aws-ecs/lib/amis.ts

```diff
@@ -280,14 +297,15 @@ export class EcsOptimizedImage implements ec2.IMachineImage {
       + (this.hwType === AmiHardwareType.NEURON ? 'inf/' : '')
       + (this.windowsVersion ? 'image_id' : 'recommended/image_id');
 
-    this.cachedInContext = props?.cachedInContext ?? false;
+    this.cachedInContext = props.cachedInContext ?? false;
+    this.additionalCacheKey = props.additionalCacheKey;
```

Comment:
We should probably add some validation to say that `additionalCacheKey` only makes sense if `cachedInContext==true` ?